lib/src/notifications/notifications_service.dart

```diff
@@ -0,0 +1,171 @@
+
+        // add the payload the the payload service
+        _payloadService.add(data);
+      }
+    } else {
+      logger.i('The app has been launched by the user');
+    }
+
+    await _flutterLocalNotificationsPlugin
+        .resolvePlatformSpecificImplementation<
+            AndroidFlutterLocalNotificationsPlugin>()
+        ?.createNotificationChannel(_buildAndroidNotificationMaxChannel());
+
+    await _flutterLocalNotificationsPlugin.initialize(
+      const InitializationSettings(
+        android: initializationSettingsAndroid,
+        iOS: initializationSettingsIOS,
+      ),
+      // this is needed to handle the tap on the notification when the app is
+      // open.
+      onDidReceiveNotificationResponse: (details) {
+        logger.i('Click on notification: $details');
+        // notifications can be remote or local.
+        // only notifications with payload are handled.
+        if (details.payload != null) {
+          final data = jsonDecode(details.payload!) as Map<String, dynamic>;
+          // if the notification is a push notification and the app has to store
+          // the notification in the db, create notification.
+
+          // add the payload the the payload service
+          _payloadService.add(data);
+        }
+      },
+    );
+  }
+
+  // Use this inside the show method
+  NotificationDetails _buildNotificationDetails(
+    StyleInformation? styleInformation,
+  ) {
+    final androidPlatformChannelSpecifics = AndroidNotificationDetails(
+      'android_ian_channel',
+      'General Notification',
+      channelDescription: 'This is a general notification channel',
+      importance: Importance.max,
+      priority: Priority.max, // high
+      showWhen: false,
+      styleInformation: styleInformation,
+    );
+
+    return NotificationDetails(
+      android: androidPlatformChannelSpecifics,
+    );
+  }
+
+  /// On Android, notification message are sent to Notification Channel which
+  /// are used to control how a notification is delivered.
+  /// The default FCM channel used is hidden from users, however provides a
+  /// "default" importance level.
+  /// Heads up notifications require a "max" importance level.
+  ///
+  /// N.B. When this method is used with FirebaseMessaging.onMessage
+  /// remember to check if message.notification?.android is not null.
+  Future<void> showNotification(ReceivedNotification notification) async {
+    // final bigPicture = _base64EncodedImage(notification.imageUrl ?? '');
+    await _flutterLocalNotificationsPlugin.show(
+      notification.id,
+      notification.title,
+      notification.body,
+      _buildNotificationDetails(
+        null,
+        // BigPictureStyleInformation(
+        //   ByteArrayAndroidBitmap.fromBase64String(bigPicture),
+        // ),
+      ),
+      payload: jsonEncode(notification.payload),
+    );
+  }
+
+  Future<void> scheduleNotification(
+    ReceivedNotification notification, {
+    required DateTime when,
```

Comment:
```suggestion
    required DateTime schedule,
```

`when` è una keyword riservata in Dart